src/driver/decode_mode.rs

```diff
@@ -24,3 +26,68 @@ impl DecodeMode {
         self != DecodeMode::Pass
     }
 }
+
+/// The channel layout of output audio when using [`DecodeMode::Decode`].
+#[derive(Clone, Copy, Debug, Default, Eq, PartialEq, Hash)]
+#[non_exhaustive]
+pub enum Channels {
+    /// Decode received audio packets into a single channel.
+    Mono,
+    /// Decode received audio packets into two interleaved channels.
+    ///
+    /// Received mono packets will automatically be duplicated across
+    /// both channels.
+    ///
+    /// The default choice.
+    #[default]
+    Stereo,
+}
+
+impl Channels {
+    pub(crate) fn n_channels(self) -> usize {
+        match self {
+            Channels::Mono => 1,
+            Channels::Stereo => 2,
+        }
+    }
+}
+
+impl From<Channels> for OpusChannels {
+    fn from(value: Channels) -> Self {
+        match value {
+            Channels::Mono => OpusChannels::Mono,
+            Channels::Stereo => OpusChannels::Stereo,
+        }
+    }
+}
+
+/// The sample rate of output audio when using [`DecodeMode::Decode`].
+#[derive(Clone, Copy, Debug, Default, Eq, PartialEq, Hash)]
+#[non_exhaustive]
+pub enum SampleRate {
+    /// Decode to a sample rate of 8kHz.
+    Hz8000,
+    /// Decode to a sample rate of 12kHz.
+    Hz12000,
+    /// Decode to a sample rate of 46kHz.
+    Hz16000,
```

Comment:
Just noticed 46kHz in the comment vs value of 16000Hz here

Reply:
Good catch!